sample_dataset/metabase/sample_dataset/generate.clj

```diff
@@ -248,6 +271,43 @@
       person
       (assoc person :orders (vec (repeatedly num-orders #(random-order person (rand-nth products))))))))
 
+(defn- add-autocorrelation
+  ([k xs] (add-autocorrelation 1 k xs))
+  ([lag k xs]
+   (map (fn [prev next]
+          (update prev k #(/ (+ % (k next)) 2)))
+        xs
+        (drop lag xs))))
+
+(defn add-increasing-variance
+  [k xs]
+  (let [n (count xs)]
+    (map-indexed (fn [i x]
+                   (update x k * (+ 1 (* (/ i n) (- (rand 2) 0.9)))))
+                 xs)))
+
+(defn add-seasonality
+  [season-fn k seasonality-map xs]
+  (for [x xs]
+    (update x k * (seasonality-map (season-fn x)))))
+
+(defn add-outliers
+  [mode n k xs]
```

Comment:
it's not super clear to me what `mode` means here. It looks like this function gets called exactly twice, once with a mode of `:share`, and once with a mode of `:count`, but only `:mode` actually ever does anything. That's something it would be good to document somewhere

Reply:
+1